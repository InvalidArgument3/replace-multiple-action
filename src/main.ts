import * as core from '@actions/core';
//import { promises as fs } from 'fs';

async function run(): Promise<void> {
  try {
    core.info('starting');
    const files: string = core.getInput('files', { required: true });
    const find: string = core.getInput('find', { required: true });

    core.info(files);
    core.info(find);
  } catch (error) {
    if (error instanceof Error) core.setFailed(error.message);
  }
}

run();
