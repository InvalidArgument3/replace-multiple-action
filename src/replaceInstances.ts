import { IReplacements } from './main';

export function replaceInstances(
  findData: IReplacements,
  content: string
): string {
  for (const pair of findData) {
    const pattern =
      // eslint-disable-next-line prefer-template
      '(^|\\s+|\\()' +
      // eslint-disable-next-line no-useless-escape
      pair.find.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&') +
      '($|\\s+|\\))';

    content = content.replace(new RegExp(pattern, 'g'), `$1${pair.replace}$2`);
  }
  return content;
}
