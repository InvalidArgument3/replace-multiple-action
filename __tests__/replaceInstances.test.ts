import { expect, test } from '@jest/globals';
import { IReplacements } from '../src/main';
import { replaceInstances } from '../src/replaceInstances';

const prefix = '(^|\\s+|\\()';
const suffix = '($|\\s+|\\))';

test('space separated', () => {
  const data: IReplacements = [
    {
      find: 'http://localhost',
      replace: 'https://wayback.org/http://localhost'
    }
  ];

  const content = 'some text http://localhost blah blah';

  const result = replaceInstances(data, content, prefix, suffix);

  expect(result).toMatchSnapshot();
});

test('start and end', () => {
  const data: IReplacements = [
    {
      find: 'http://localhost',
      replace: 'https://wayback.org/http://localhost'
    }
  ];

  const content = 'http://localhost blah \nblah and http://localhost';

  const result = replaceInstances(data, content, prefix, suffix);

  expect(result).toMatchSnapshot();
});

test('in a link', () => {
  const data: IReplacements = [
    {
      find: 'http://localhost',
      replace: 'https://wayback.org/http://localhost'
    }
  ];

  const content = 'some [text](http://localhost) blah blah';

  const result = replaceInstances(data, content, prefix, suffix);

  expect(result).toMatchSnapshot();
});

test('not links', () => {
  const data: IReplacements = [
    {
      find: 'http://localhost',
      replace: 'https://wayback.org/http://localhost'
    }
  ];

  const content =
    'some [text]http://localhost) blah blah and this -http://localhost';

  const result = replaceInstances(data, content, prefix, suffix);

  expect(result).toMatchSnapshot();
});
