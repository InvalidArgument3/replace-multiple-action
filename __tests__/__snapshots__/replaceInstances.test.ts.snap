// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`in a link 1`] = `"some [text](https://wayback.org/http://localhost) blah blah"`;

exports[`not links 1`] = `"some [text]http://localhost) blah blah and this -http://localhost"`;

exports[`space separated 1`] = `"some text https://wayback.org/http://localhost blah blah"`;

exports[`start and end 1`] = `
"https://wayback.org/http://localhost blah 
blah and https://wayback.org/http://localhost"
`;
